UPGRADE.md

```diff
@@ -30,13 +30,73 @@ mysqldump -u YOUR_ILIOS_DATABASE_USERNAME -h YOUR_ILIOS_HOSTNAME -R -p YOUR_ILIO
+
+Good job migrating your database, but we're not quite finished yet!
+
+## Migrating your old learning materials and cleaning up the text of your Topics
+
+### Learning Materials
+The file storage/naming functionality for handling Learning Materials in Ilios 3 has changed slightly from the way it was done in Ilios 2.x.  In order to support the new formatting and storage, you will need to migrate your learning materials to their new storage location and naming convention.  Thankfully, Ilios 3 provides a console command to do this!
+
+The Learning Material migration *copies* learning materials, it does not *move* them so, before you begin migrating your learning materials, you want to make ABSOLUTELY CERTAIN that you have enough free space in your system's storage (hard disk space) in order to accommodate copying all of your materials to the new location.  At the very least, you will want as much free space as you already use for your learning materials.
+
+To find out how much space your learning materials are currently taking, login to your Ilios server and change directories to the folder that contains your learning materials and then run this command:
+
+```bash
+df -h . 
+```
+
+This is will output something like the following:
+```bash
+Filesystem   Size   Used  Avail Capacity  iused    ifree %iused  Mounted on
+/dev/disk1  465Gi  209Gi  255Gi    46% 54901683 66942027   45%   /
+```
+
+If the total number of Gigabytes listed under 'Avail' is significantly greater (20Gi+) than what is listed under 'Used,' you should be good to go!  If the 'Avail' value is less (or even just "pretty close") to what is listed under 'Used', you will probably want to increase the storage on your server before running the Learning Material migration.  If you try to run this migration and do not have enough space to store the files when they are copied over, it can fill up your storage space completely and render your server inoperable!  Please be careful and be certain of your storage availability before performing this operation!
+
+If you have plenty of space and believe you are able to migrate your learning materials, do the following:
+
+1. Backup your learning materials!!!
+2. Seriously! Back up your learning materials before running this command!!!  If you cannot back up your learning materials or quarantine a copy of them to a separate place on your server where they cannot be affected by this process, you should really consider backing them up offline to some type of external storage device.  Do not risk losing your existing learning materials!
+3. Now that you're sure you're ready to migrate, navigate into your Ilios application's root directory (eg, /web/ilios3/ilios) and examine the file at app/config/parameters.yml.  Look for the value that corresponds to file_system_storage_path.  This is where your learning materials will be moved.
```

Comment:
Change your learning materials will be moved to 'your learning materials will be copied to'